### PR TITLE
Add support for yarn-berry lockfiles

### PR DESCRIPTION
**Pull request by mikeduminy**
Added support for parsing yarn-berry lockfiles (they are pure yaml).

We can no longer look up dependencies by `package@version` because berry adds additional things like a protocol (e.g. `package@protocol:version`), so there's now a regex (with tests) that strips out the protocol.

An argument could be made that this should be a separate command, like npm and yarn. Let me know what feels best.

Also, maybe this could be a berry plugin? I'm not sure what the capabilities are.

### CODE REVIEW

**Comment by oblador** [comment]
Been a whike since i looked into berry, but yes i think it can be solved to some degree with config instead, although at potentially higher maintenance cost. 

Btw, have you looked into this package that seems to support both formats? https://www.npmjs.com/package/@yarnpkg/parsers

**Comment by mikeduminy** [comment]
That works even better, good idea!

File: test/parse-yarn-descriptor.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  it('should parse a yarn berry descriptor with protocol', () => {
+    expect(parseYarnDescriptor('hest@npm:1.0.0')).toEqual({
+      packageName: 'hest',
+      version: '1.0.0',
+    });
+  });
+
+  it('should parse a yarn berry descriptor with scope', () => {
+    expect(parseYarnDescriptor('@snel/hest@1.0.0')).toEqual({
+      packageName: '@snel/hest',
```
Comment (oblador) [inline test/parse-yarn-descriptor.spec.js:20]: ⭐

File: src/parseYarnDescriptor.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Supported formats:
+ * - `package@version`
+ * - `@scope/package@version`
+ * - `package@protocol:version`
+ * - `@scope/package@protocol:version`
+ *
+ * @param {string} descriptor - The yarn berry descriptor to parse.
+ * @returns {ParsedDescriptor} The parsed descriptor.
+ */
+function parseYarnDescriptor(descriptor) {
```
Comment (oblador) [inline src/parseYarnDescriptor.js:32]:
```suggestion
function parseBerryDescriptor(descriptor) {
```
Maybe? I was worried this regex would slow down non-berry users at first glance. Speaking of that, how's the perf with a large berry workspace like yours?

**Review by oblador — APPROVED** [review]
Worth mentioning in README that it supports berry maybe?

File: src/parseYarnDescriptor.js (hunk 2 above)
Comment (mikeduminy) [reply src/parseYarnDescriptor.js:32]: Performance looks good. Only adds ~4% increase in time in a large project

Before PR:
```
  Time (mean ± σ):      4.028 s ±  0.101 s    [User: 3.634 s, System: 0.342 s]
  Range (min … max):    3.935 s …  4.207 s    10 runs
```
After PR:
```
  Time (mean ± σ):      4.121 s ±  0.152 s    [User: 3.748 s, System: 0.328 s]
  Range (min … max):    3.975 s …  4.453 s    10 runs
```

Comment (mikeduminy) [reply src/parseYarnDescriptor.js:32]: As for the suggestion, this function parses both yarn v1 and berry descriptors. At the time of calling it we don't know which version of yarn lockfile we are parsing. Unless of course we interpret the "metadata" value in the parse result, but that's a bit messy.

I can clarify that this parses both descriptors in the JSDoc

File: package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
   },
   "homepage": "https://github.com/oblador/diglett#readme",
   "dependencies": {
-    "@yarnpkg/lockfile": "^1.1.0",
+    "@yarnpkg/parsers": "^3.0.0-rc.48.1",
```
Comment (oblador) [inline package.json:40]:
```suggestion
    "@yarnpkg/parsers": "^2.5.1",
```

File: src/parseYarnDescriptor.js (hunk 2 above)
Comment (oblador) [reply src/parseYarnDescriptor.js:37]: Sweet, makes sense!

**Comment by oblador** [comment]
@mikeduminy Need to update lock file too.

**Comment by mikeduminy** [comment]
@oblador done

**Pull request merged**